statefun-sdk-js/src/statefun.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+"use strict";
+
+import {FunctionSpec, ValueSpec, Address, Type, FunctionOpts} from "./core";
+import {Context} from "./context";
+import {Message, messageBuilder, egressMessageBuilder} from "./message";
+import {kafkaEgressMessage, kinesisEgressMessage} from "./egress";
+import {handle} from "./handler";
+
+import {BOOL_TYPE, CustomType, FLOAT_TYPE, INT_TYPE, JsonType, ProtobufType, STRING_TYPE} from "./types";
+
+class StateFun {
+    readonly #fns: Record<string, FunctionSpec>;
+
+    constructor() {
+        this.#fns = {};
+    }
+
+    bind(opts: FunctionOpts) {
+        const spec = FunctionSpec.fromOpts(opts);
+        this.#fns[spec.typename] = spec;
+    }
+
+    static intType(): Type<number> {
```

Comment:
Are all of these static functions well-placed on the StateFun class? Why not just offer them standalone? That would reduce the footprint of this class.